linux/check_os_fullname/validate_os_fullname.yml

```diff
@@ -1,63 +1,95 @@
 # Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
 ---
-# Wait VMware tools to report Guest OS fullname
-- include_tasks: ../../common/vm_get_guest_info.yml
-
-# If the guest os full name is unmapped and displays OS detailed data, test passed
-- block:
-    - name: "guest OS detailed information is shown in guest OS full name, test passed"
-      ansible.builtin.meta: end_host
-  when:
-    - (esxi_version is version('6.7.0', '==') and (esxi_update_version | int < 2)) or (esxi_version is version('7.0.0', '=='))
-    - unmapped_os_fullname is defined and unmapped_os_fullname
-    - unmapped_os_fullname in guestinfo_guest_full_name
-
-- name: "Check guest ID in guest info is not empty"
+- name: "Get VM's guest info"
+  include_tasks: ../../common/vm_get_guest_info.yml
+
+- name: "Display ESXi version and VM's hardware version"
+  ansible.builtin.debug:
+    msg:
+      - "ESXi version: {{ esxi_version }}"
+      - "VM's hardware version: {{ vm_hardware_version }}"
+
+# If the guest os full name is unmapped and displays OS detailed data on ESXi 6.7 GA and Update 1,
+# or on ESXi 7.0.0, that's expected. Test passed.
+- name: "Check unmapped guest OS full name"
+  block:
+    # The unmapped guest full name which at least includes Linux kernel version, OS pretty name
+    # For VM with newer VMware Tools, the unmapped guest full name also includes OS name, version id and build id
+    # Here we only check linux kernel version and OS pretty name
+    - name: "Set unmapped guest full name for {{ vm_guest_os_distribution }} on ESXi {{ esxi_version }}"
+      ansible.builtin.set_fact:
+        unmapped_guest_fullname: "Linux {{ guest_os_ansible_kernel }} {{ guest_os_release.PRETTY_NAME }}"
+      when: guest_os_release.PRETTY_NAME | default('')
+
+    - name: "Test passed"
+      block:
+        - name: "Guest OS full name is unmapped"
+          ansible.builtin.debug:
+            msg: >-
+              Guest id is '{{ guestinfo_guest_id }}' on ESXi {{ esxi_version }}, and
+              guest OS full name is '{{ guestinfo_guest_full_name }}',  which shows
+              guest OS detailed information and is as expected on ESXi {{ esxi_version }}.
+              Test Passed.
+        - name: "Test passed on ESXi {{ esxi_version }}"
+          ansible.builtin.meta: end_host
+      when:
+        - unmapped_guest_fullname is defined and unmapped_guest_fullname
+        - unmapped_guest_fullname in guestinfo_guest_full_name
+  when: >
```

Comment:
It seems here is missing one parameter when guest OS is otherlinux

Reply:
There is no OS in our [compatible GOS list](https://github.com/vmware/ansible-vsphere-gos-validation#compatible-guest-operating-systems) will be mapped to `otherLinuxGuest` or `otherLinux64Guest`. If it does, it should be a product bug.

Reply:
I'm still confused about the logic here. E.g. if we run the case with rhe9 on ESXi 7.0.0, will this block be executed?